flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java

```diff
@@ -1269,6 +1271,112 @@ public void testAlterTable() throws Exception {
+                                .columnByMetadata("m", DataTypes.INT(), true)
+                                .columnByExpression("g", "TO_TIMESTAMP(e)")
+                                .watermark("f", "g - INTERVAL '5' SECOND")
+                                .primaryKey("a", "b")
+                                .build(),
+                        "tb1",
+                        Collections.emptyList(),
+                        Collections.emptyMap());
+        catalogManager.setCurrentCatalog("cat1");
+        catalogManager.setCurrentDatabase("db1");
+        catalog.createTable(new ObjectPath("db1", "tb1"), catalogTable, true);
+        // Test alter table rename
+        // rename column b
+        Operation operation = parse("alter table tb1 rename b to b1", SqlDialect.DEFAULT);
+        assert operation instanceof AlterTableSchemaOperation;
+        Schema actual =
+                ((AlterTableSchemaOperation) operation).getCatalogTable().getUnresolvedSchema();
+        Schema expected =
+                Schema.newBuilder()
+                        .column("a", DataTypes.STRING().notNull())
+                        .column("b1", DataTypes.INT().notNull())
+                        .column("c", DataTypes.INT())
+                        .column("d", DataTypes.INT())
+                        .column("e", DataTypes.STRING())
+                        .column("f", DataTypes.TIMESTAMP(3).notNull())
+                        .columnByExpression("h", "c - 1")
+                        .columnByExpression("i", "cat1.`default`.my_udf1(d) + 1")
+                        .columnByMetadata("m", DataTypes.INT(), true)
+                        .columnByExpression("g", "TO_TIMESTAMP(e)")
+                        .watermark("f", "g - INTERVAL '5' SECOND")
+                        .primaryKeyNamed("PK_a_b", "a", "b1")
+                        .build();
+        assertThat(expected).isEqualTo(actual);
+
+        // rename column c test computed column case1
+        assertThatThrownBy(() -> parse("alter table tb1 rename c to c1", SqlDialect.DEFAULT))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "Old column c is referenced by computed column `h` INT AS c - 1, "
+                                + "currently doesn't allow to rename column which is referenced by computed column.");
+
+        // rename column d test computed column case2
+        assertThatThrownBy(() -> parse("alter table tb1 rename d to d1", SqlDialect.DEFAULT))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "Old column d is referenced by computed column `i` INT NOT NULL AS cat1.`default`.my_udf1(d) + 1,"
+                                + " currently doesn't allow to rename column which is referenced by computed column.");
+
+        // rename column e test computed column case3
+        CatalogTable catalogTable2 =
```

Comment:
Can we test the computed column case3 with `tb1` so that the code can be more readable.

Reply:
This case in order to validate the computed column `**j**` is created by `ApiExpression`, this class doesn't implement the `equals` method, so we can't reuse the `tb1`, otherwise, the before test will fail.